### PR TITLE
Add workflow with shellcheck to do some sanity checks on shell scripts

### PR DESCRIPTION
**Pull request by marcofranssen**

### CODE REVIEW

File: .github/workflows/scripts/update-versions.sh [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
+  echo "  repo: ${REPO_URL}"
+  echo "  current version: ${VERSION}"
+  helm repo add "${CHART}" "${REPO_URL}" > /dev/null
+  helm repo update "${CHART}" > /dev/null
+  LATEST_VERSION=$(helm search repo --regexp "${CHART}/${CHART}\v" -o json | jq -r '.[0].version')
+  echo "  latest version: ${LATEST_VERSION}"
+  if [ "x${VERSION}" != "x${LATEST_VERSION}" ]; then
     echo "  New version found!"
-    jq "( $ENTRYQUERY ).version |= "'"${LATEST_VERSION}"' "${CHARTJSON}" > /tmp/$$
+    jq "(${ENTRYQUERY}).version |= "'"${LATEST_VERSION}"' "${CHARTJSON}" > /tmp/$$
```
Comment (kfox1111) [inline .github/workflows/scripts/update-versions.sh:21]: The quoting here is still wrong.

File: .github/tests/namespace-override/post-install.sh [hunk 2]
```diff
@@ -13,7 +14,7 @@ k_rollout_status () {
   kubectl rollout status --watch --timeout 30s --namespace "$1" "$2" "$3" | tail -n 1
 }
 
-RELEASE=$(helm ls --no-headers -n "${scenario}" | awk '{print $1}' | grep 'spire-[^-]*$')
+RELEASE=$(helm ls --no-headers -n "${NS}" | awk '{print $1}' | grep 'spire-[^-]*$')
```
Comment (kfox1111) [inline .github/tests/namespace-override/post-install.sh:17]: Where is NS coming from?

File: .github/tests/namespace-override/post-install.sh [hunk 3]
```diff
@@ -30,7 +31,7 @@ cat <<EOF >>"$GITHUB_STEP_SUMMARY"
 | spire-spiffe-oidc-discovery-provider | <pre>$(k_wait spire-server deployments.apps "${RELEASE}-spiffe-oidc-discovery-provider")</pre> |
 EOF
 
-if [ $1 -ne 0 ]; then
+if [ "$1" -ne '0' ]; then
```
Comment (kfox1111) [inline .github/tests/namespace-override/post-install.sh:34]: I don't understand this one. != is used for strings, -ne for integers.

File: .github/tests/production-example/post-install.sh [hunk 4]
```diff
@@ -38,14 +44,14 @@ cat <<EOF >>"$GITHUB_STEP_SUMMARY"
...
-| spire-agent                          | "$("${k_rollout_status[@]}" spire-system daemonset spire-agent)" |
+| spire-server                         | "$(k_rollout_status spire-server statefulset spire-server)" |
+| spire-controller-manager             | "$(k_rollout_status spire-server statefulset spire-controller-manager)" |
+| spire-spiffe-oidc-discovery-provider | "$(k_wait spire-server deployments.apps spire-spiffe-oidc-discovery-provider)" |
+| spire-spiffe-csi-driver              | "$(k_rollout_status spire-system daemonset spire-spiffe-csi-driver)" |
+| spire-agent                          | "$(k_rollout_status spire-system daemonset spire-agent)" |
 EOF
 
-if [ $1 -ne 0 ]; then
+if [ "$1" -ne '0' ]; then
```
Comment (kfox1111) [inline .github/tests/production-example/post-install.sh:54]: same

**Review by kfox1111 — COMMENTED** [review]
Most of this looks good. A couple of issues.

File: .github/workflows/scripts/update-versions.sh (hunk 1 above)
Comment (marcofranssen) [reply .github/workflows/scripts/update-versions.sh:21]: Yes, correct will resolve that in that other PR once this is merged, that other PR will have a much smaller diff.

File: .github/tests/namespace-override/post-install.sh (hunk 3 above)
Comment (marcofranssen) [reply .github/tests/namespace-override/post-install.sh:34]: woops good catch

File: .github/tests/prometheus/post-install.sh [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+set -x
+
+SCRIPT=$(readlink -f "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+scenario="${SCRIPTPATH##*/}"
```
Comment (kfox1111) [inline .github/tests/prometheus/post-install.sh:7]: This line is inconsistent then some of the rest? Should they all be the same?

File: .github/tests/production-example/post-install.sh [hunk 6]
```diff
@@ -38,14 +44,14 @@ cat <<EOF >>"$GITHUB_STEP_SUMMARY"
 
 | workload                             | Status |
 | ------------------------------------ | ------ |
-| spire-server                         | "$("${k_rollout_status[@]}" spire-server statefulset spire-server)" |
-| spire-controller-manager             | "$("${k_rollout_status[@]}" spire-server statefulset spire-controller-manager)" |
-| spire-spiffe-oidc-discovery-provider | "$("${k_wait[@]}" spire-server deployments.apps spire-spiffe-oidc-discovery-provider)" |
-| spire-spiffe-csi-driver              | "$("${k_rollout_status[@]}" spire-system daemonset spire-spiffe-csi-driver)" |
-| spire-agent                          | "$("${k_rollout_status[@]}" spire-system daemonset spire-agent)" |
+| spire-server                         | "$(k_rollout_status spire-server statefulset spire-server)" |
```
Comment (kfox1111) [inline .github/tests/production-example/post-install.sh:47]: Does this need the pre stuff?

**Review by kfox1111 — COMMENTED** [review]
Couple more things

File: .github/tests/production-example/post-install.sh (hunk 6 above)
Comment (marcofranssen) [reply .github/tests/production-example/post-install.sh:47]: good catch

Comment (marcofranssen) [reply .github/tests/production-example/post-install.sh:47]: this is resolved in #88

**Review by kfox1111 — APPROVED** [review]
Close enough. It makes things way better and future pr's already proposed will fix the rest.
LGTM

**Pull request merged**